### ISSUE TITLE
Block until all `rayon::spawn` have completed

### ISSUE THREAD

**Issue by gyscos** [rayon-rs/rayon#650]
Until we have a way to parallelize a generic non-Send iterator, I'm using `rayon::spawn` from a single-threaded loop.
This works, except that the main thread will not wait for the tasks to complete before exiting.
Is there a way to block until all global tasks have been completed? Or should I maintain my own completion count for that?

**Comment by cuviper** [comment]
We don't currently have a way to wait for idle, but I think we probably should.

**Comment by wagnerf42** [comment]
how about using a scope ?

**Comment by cuviper** [comment]
Indeed, a `scope` will not return until all of its `Scope::spawn`s have completed. However, it does require that the initial closure is `Send` (#562), so @gyscos would have to create their non-`Send` iterator from within. And since this consumes one thread of the pool, you probably want to make sure there are more threads to do the spawned work, otherwise the whole iterator will be processed before getting to the spawns.

**Comment by gsquire** [comment]
Perhaps you could leverage a [WaitGroup](https://docs.rs/crossbeam/0.7.1/crossbeam/sync/struct.WaitGroup.html)?

**Comment by cuviper** [comment]
Note: we do have `in_place_scope` now, without `Send`, so you can `spawn` tasks from the current thread and have a well-defined wait at the end of that scope.
